partiql-tests-data-extended/eval/primitives/functions/text_replace.ion

```diff
@@ -0,0 +1,228 @@
+      output:"abc😁😞abc😁😞"
+    }
+  },
+  {
+    name:"text_replace('abc😁😞abc😁😞', '😁😞', 'def')",
+    statement:"text_replace('abc😁😞abc😁😞', '😁😞', 'def')",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[EvalModeCoerce, EvalModeError],
+      output:"abcdefabcdef"
+    }
+  },
+  {
+    name:"text_replace('abcdefabcdef', 'def', 'd😁😞')",
+    statement:"text_replace('abcdefabcdef', 'def', 'd😁😞')",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[EvalModeCoerce, EvalModeError],
+      output:"abcd😁😞abcd😁😞"
+    }
+  },
+  {
+    name:"text_replace('abcdefabcdef', 'def', 'deࠫf')",
+    statement:"text_replace('abcdefabcdef', 'def', 'deࠫf')",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[EvalModeCoerce, EvalModeError],
+      output:"abcdeࠫfabcdeࠫf"
+    }
+  },
+  {
+    name:"text_replace('abcdeࠫfabcdeࠫf', 'def', 'XX')",
+    statement:"text_replace('abcdeࠫfabcdeࠫf', 'def', 'XX')",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[EvalModeCoerce, EvalModeError],
+      output:"abcdeࠫfabcdeࠫf"
+    }
+  },
+  {
+    name:"text_replace('abcdeࠫfabcdeࠫf', 'deࠫf', 'XX')",
+    statement:"text_replace('abcdeࠫfabcdeࠫf', 'deࠫf', 'XX')",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[EvalModeCoerce, EvalModeError],
+      output:"abcXXabcXX"
+    }
+  },
+  {
+    name:"text_replace NULL first arg",
+    statement:"text_replace(NULL, 'foo', 'bar')",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[EvalModeCoerce, EvalModeError],
+      output:null
+    }
+  },
+  {
+    name:"text_replace NULL second arg",
+    statement:"text_replace('foo', NULL, 'bar')",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[EvalModeCoerce, EvalModeError],
+      output:null
+    }
+  },
+  {
+    name:"text_replace NULL third arg",
+    statement:"text_replace('foo', 'bar', NULL)",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[EvalModeCoerce, EvalModeError],
+      output:null
+    }
+  },
+  {
+    name:"text_replace MISSING first arg",
+    statement:"text_replace(MISSING, 'foo', 'bar')",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[EvalModeCoerce, EvalModeError],
+      output:$missing::null
+    }
+  },
+  {
+    name:"text_replace MISSING second arg",
+    statement:"text_replace('foo', MISSING, 'bar')",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[EvalModeCoerce, EvalModeError],
+      output:$missing::null
+    }
+  },
+  {
+    name:"text_replace MISSING third arg",
+    statement:"text_replace('foo', 'bar', MISSING)",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[EvalModeCoerce, EvalModeError],
+      output:$missing::null
+    }
+  },
+  {
+    name:"text_replace NULL and MISSING arg",
+    statement:"text_replace(NULL, 'foo', MISSING)",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[EvalModeCoerce, EvalModeError],
+      output:$missing::null
+    }
```

Comment:
`MISSING` over `NULL` propagation isn't discussed in the PartiQL spec. I had previously created an issue to track this: https://github.com/partiql/partiql-spec/issues/35. I discussed with Almann about this previously and we determined that `MISSING` should "win over" `NULL`. Perhaps I forgot to followup on the issue. There are some existing tests for this behavior in the non-extended directory. E.g. https://github.com/partiql/partiql-tests/blob/95e16c5018ae384bc010c232c43215928823e05a/partiql-tests-data/eval/primitives/functions/substring.ion#L1178-L1189

`partiql-lang-kotlin` in permissive mode follows this behavior.

> I guess a rationale could be that since MISSING is the permissive mode's stand-in for an error and the error would "win" over NULL in the strict mode?

That's the way I had generally viewed why `MISSING` propagates over `NULL`.

> would this new principle force us to reconsider other NULL vs MISSING design decisions?

I'm not aware of any contradictions or confusing behavior due to this decision.